client/src/components/EventCalendar/index.tsx

```diff
@@ -0,0 +1,152 @@
+import { FC, useState, useEffect } from 'react'
+import FullCalendar, { formatDate } from '@fullcalendar/react'
+import dayGridPlugin from '@fullcalendar/daygrid'
+import timeGridPlugin from '@fullcalendar/timegrid'
+import interactionPlugin from '@fullcalendar/interaction'
+import './style.css'
+import { useNavigate } from 'react-router-dom'
+import ApiService from '../../services/ApiService'
+
+const EventCalendar:FC = () => {
+  const [initialEvents, setInitialEvents] = useState([])
+  const [interestedEvents, setInterestedEvents] = useState([])
+  const navigate = useNavigate()
+  const handleEventClick = (clickInfo:any):void => {
+    const eventId = clickInfo?.event?.extendedProps.eventId
+
+    if (window.confirm(`Are you sure you want to go to the event '${clickInfo.event.title}'`)) {
+      navigate(`/event/${eventId}`)
+    }
+  }
+
+  const renderEventContent = (eventInfo:any):any => {
+    const status = eventInfo?.event.extendedProps.status
+    const type = eventInfo?.event.extendedProps.type
+    return (
+      <div style={{
+        flexDirection: 'column',
+        justifyContent: 'center',
+        alignItems: 'center',
+        textAlign: 'center',
+        margin: 'auto',
+      }}
+      >
+        <h3 style={{ textTransform: 'capitalize', fontSize: '0.8rem' }}>{type}</h3>
+        <p className="calendar-event-info">{eventInfo.timeText}</p>
+        <p
+          className="calendar-event-info"
+          style={{ fontSize: '1rem', fontWeight: 500 }}
+        >
+          {eventInfo.event.title}
+
+        </p>
+        <p className={`event-status ${status}`}>{status}</p>
+      </div>
+    )
+  }
+
+  const renderSidebarEvent = (event:any):any => (
```

Comment:
remove `any`
and you can move this to other file and make it as a react component with `props`